### PR TITLE
ncm-systemd: Add support for mount, automount and timer units

### PR DESCRIPTION
**Pull request by wpoely86**
Add support for:
-  `[Mount]` and `.mount` units
-  `[Automount]` and `.automount` units
-  `[Timer]` and `.timer` units

in ncm-systemd.

### CODE REVIEW

**Review by ned21 — APPROVED** [review]

File: ncm-systemd/src/main/pan/components/systemd/schema.pan [hunk 1]
```diff
@@ -321,6 +321,57 @@ type ${project.artifactId}_unitfile_config_socket = {
...
+the [Timer] section
+http://www.freedesktop.org/software/systemd/man/systemd.timer.html
+}
+type ${project.artifactId}_unitfile_config_timer = {
+    'OnActiveSec' ? long(0..)
+    'OnBootSec' ? long(0..)
+    'OnStartupSec' ? long(0..)
+    'OnUnitActiveSec' ? long(0..)
+    'OnUnitInactiveSec' ? long(0..)
+    'OnCalendar' ? string[]
```
Comment (wpoely86) [inline ncm-systemd/src/main/pan/components/systemd/schema.pan:363]: I'm not sure what to do with this one. It supports a lot of different syntax: https://www.freedesktop.org/software/systemd/man/systemd.time.html#

File: ncm-systemd/src/main/pan/components/systemd/schema.pan [hunk 2]
```diff
@@ -375,7 +429,7 @@ type ${project.artifactId}_target = string with match(SELF, "^(default|poweroff|
 type ${project.artifactId}_unit_type = {
     "name" ? string # shortnames are ok; fullnames require matching type
```
Comment (wpoely86) [inline ncm-systemd/src/main/pan/components/systemd/schema.pan:430]: Regarding this comment, is there a reason why it's not enforced by validation @stdweird ?

Comment (stdweird) [reply ncm-systemd/src/main/pan/components/systemd/schema.pan:430]: because it's easier not to? feel free to add it, shouldn't be that hard

Comment (wpoely86) [reply ncm-systemd/src/main/pan/components/systemd/schema.pan:430]: I'm more thinking about, if you have `data-mnt.mount` as filename, the `type` should match and if you have a file, it should have `mount` section.

Comment (wpoely86) [reply ncm-systemd/src/main/pan/components/systemd/schema.pan:430]: I'm actually not sure what the `name` entry here does?

Comment (stdweird) [reply ncm-systemd/src/main/pan/components/systemd/schema.pan:430]: i guess it forces the name?

File: ncm-systemd/src/main/pan/components/systemd/schema.pan (hunk 1 above)
Comment (jrha) [reply ncm-systemd/src/main/pan/components/systemd/schema.pan:397]: Oh wow.

Either leave it as is, or pick a subset of the supported syntax to validate.

File: ncm-systemd/src/main/pan/components/systemd/schema.pan (hunk 2 above)
Comment (jrha) [reply ncm-systemd/src/main/pan/components/systemd/schema.pan:464]: Are you using it?

File: ncm-systemd/src/main/pan/components/systemd/schema.pan (hunk 1 above)
Comment (wpoely86) [reply ncm-systemd/src/main/pan/components/systemd/schema.pan:397]: OK, leave as string.

File: ncm-systemd/src/main/pan/components/systemd/schema.pan (hunk 2 above)
Comment (wpoely86) [reply ncm-systemd/src/main/pan/components/systemd/schema.pan:464]: No

Comment (jrha) [reply ncm-systemd/src/main/pan/components/systemd/schema.pan:464]: Probably best to remove it then.

Comment (wpoely86) [reply ncm-systemd/src/main/pan/components/systemd/schema.pan:464]: OK but that might break templates for some people? Better deprecated it first?

Comment (stdweird) [reply ncm-systemd/src/main/pan/components/systemd/schema.pan:464]: why remove it? adding the type check should be easy to do.

Comment (jrha) [reply ncm-systemd/src/main/pan/components/systemd/schema.pan:464]: Sorry I misread the PR and thought it added it, it can be left as is then.

**Review by jrha — APPROVED** [review]

**Pull request merged**